sklearn/utils/_seq_dataset.pyx.tp

```diff
@@ -197,19 +197,22 @@ cdef class SequentialDataset{{name_suffix}}:
                      current_index)
 
         # transform the pointed data in numpy CSR array
-        cdef cnp.ndarray[{{c_type}}, ndim=1] x_data = np.empty(nnz,
-                                                              dtype={{np_type}})
-        cdef cnp.ndarray[int, ndim=1] x_indices = np.empty(nnz, dtype=np.int32)
-        cdef cnp.ndarray[int, ndim=1] x_indptr = np.asarray([0, nnz],
-                                                           dtype=np.int32)
+        cdef {{c_type}}[:] x_data = np.empty(nnz, dtype={{np_type}})
+        cdef int[:] x_indices = np.empty(nnz, dtype=np.int32)
+        cdef int[:] x_indptr = np.asarray([0, nnz], dtype=np.int32)
 
         for j in range(nnz):
             x_data[j] = x_data_ptr[j]
             x_indices[j] = x_indices_ptr[j]
 
         cdef int sample_idx = self.index_data_ptr[current_index]
 
-        return (x_data, x_indices, x_indptr), y, sample_weight, sample_idx
+        return (
+            (x_data.base, x_indices.base, x_indptr.base),
```

Comment:
After discussing with @jeremiedbb, this is safer, more explicit and not more costly (should be a not open if the memory-view is not sub-strided.
```suggestion
            (np.asarray(x_data), np.asarray(x_indices), np.asarray(x_indptr),
```

This also applies in other places.